### DISCUSSION TITLE
How to set the database?

### DISCUSSION THREAD

**Discussion by pluc** [doctrine/doctrine2#9051] [Support Questions] [answered]
Hey. Simple question, but can't find ANY documentation on this anywhere, which is super weird.

How do you tell the ORM to use the database named "whatever"?? Normally (eg with PDO) I would expect to pass that in the connection settings, but doing so - using the same format as PDO, doesn't set the database name and so I'm left with a bunch of exceptions because I have "No database selected"...

**Comment by greg0ire** [comment]
The connection is not the concern of the ORM, maybe that's why you could not find docs about it? Here are the DBAL docs about this: https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/configuration.html#configuration

**Comment by pluc** [comment]
Is it the DBAL that provisions the database based on the Entity models? Cause from where I'm sitting it looks like it should be the ORM itself, and when I run vendor/bin/doctrine `orm:schema-tool:create`, does it have anything to do with the DBAL? That's what is giving me the error, not the DBAL...

**Comment by pluc** [comment, marked as answer]
Found my answer (and it doesn't involve DBAL):

The parameter I had for "database name" in the configuration array passed to `EntityManager::create()` was using `name` instead of `dbname`.

**Reply by greg0ire** [reply]
It does involve the DBAL: https://github.com/doctrine/orm/blob/2362aa1a7a43d1fb998d32c5c920371d206a00b2/lib/Doctrine/ORM/EntityManager.php#L927 (`DriverManager` is a class from the DBAL). `dbname` is mentioned in the documentation linked above.
